up_transport_zenoh/zenohutils.py

```diff
@@ -72,77 +70,71 @@ def get_uauth_from_uuri(uri: UUri) -> Union[str, UStatus]:
     @staticmethod
     def to_zenoh_key_string(authority_name: str, src_uri: UUri, dst_uri: UUri = None) -> str:
         src = ZenohUtils.uri_to_zenoh_key(authority_name, src_uri)
-        dst = ZenohUtils.uri_to_zenoh_key(authority_name, dst_uri) if dst_uri else "{}/{}/{}/{}"
+        dst = ZenohUtils.uri_to_zenoh_key(authority_name, dst_uri) if dst_uri and dst_uri != UUri() else "{}/{}/{}/{}"
         return f"up/{src}/{dst}"
 
     @staticmethod
     def map_zenoh_priority(upriority: UPriority) -> Priority:
         mapping = {
-            UPriority.UPRIORITY_CS0: Priority.BACKGROUND(),
-            UPriority.UPRIORITY_CS1: Priority.DATA_LOW(),
-            UPriority.UPRIORITY_CS2: Priority.DATA(),
-            UPriority.UPRIORITY_CS3: Priority.DATA_HIGH(),
-            UPriority.UPRIORITY_CS4: Priority.INTERACTIVE_LOW(),
-            UPriority.UPRIORITY_CS5: Priority.INTERACTIVE_HIGH(),
-            UPriority.UPRIORITY_CS6: Priority.REAL_TIME(),
-            UPriority.UPRIORITY_UNSPECIFIED: Priority.DATA_LOW(),
+            UPriority.UPRIORITY_CS0: Priority.BACKGROUND,
+            UPriority.UPRIORITY_CS1: Priority.DATA_LOW,
+            UPriority.UPRIORITY_CS2: Priority.DATA,
+            UPriority.UPRIORITY_CS3: Priority.DATA_HIGH,
+            UPriority.UPRIORITY_CS4: Priority.INTERACTIVE_LOW,
+            UPriority.UPRIORITY_CS5: Priority.INTERACTIVE_HIGH,
+            UPriority.UPRIORITY_CS6: Priority.REAL_TIME,
+            UPriority.UPRIORITY_UNSPECIFIED: Priority.DATA_LOW,
         }
         return mapping[upriority]
 
-    @staticmethod
-    def to_upayload_format(encoding: Encoding) -> UPayloadFormat:
-        try:
-            value = int(encoding.suffix)
-            return value if UPayloadFormat.Name(value) else None
-        except (ValueError, AttributeError):
-            return None
-
     @staticmethod
     def uattributes_to_attachment(uattributes: UAttributes):
-        attachment = [("", UATTRIBUTE_VERSION.to_bytes(1, byteorder='little')), ("", uattributes.SerializeToString())]
-        return attachment
+        # Convert the version number to bytes (assuming 1 as in the Rust example)
+        version_bytes = UATTRIBUTE_VERSION.to_bytes(1, byteorder='little')
+
+        # Serialize the UAttributes to bytes
+        uattributes_bytes = uattributes.SerializeToString()
+
+        # Combine version bytes and uattributes bytes into one list of bytes
+        attachment_bytes = version_bytes + uattributes_bytes
```

Comment:
We can use `attachment_bytes = [version_bytes, uattributes_bytes]` and ZBytes will transform the list.